### PR TITLE
"Native-setup-project-service"

### PR DESCRIPTION
**Pull request by MallanagoudaB**
# Description
These recommendations are intended to promote code quality and team communication during software development. They cover a variety of topics, including ensuring that pull requests are submitted to the correct branch, documenting new methods, preserving consistency across many services, and avoiding typical blunders like accessing APIs or DB queries within loops. Sensitive data should not be uploaded, and changes to environment variables or database models should be executed consistently. Teams may work more effectively and develop higher-quality software by adhering to these standards.

## Type of change
Please choose appropriate options.

- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  New feature (non-breaking change which adds functionality)
- [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ]  Enhancement (additive changes to improve performance)
- [ ]  This change requires a documentation update

## Checklist 

- [ ]  It's critical to avoid making needless file modifications in contributions, such as adding new lines, console logs, or additional spaces, to guarantee cleaner and more efficient code. Furthermore, eliminating unnecessary imports from a file might enhance code readability and efficiency.
- [ ]  Ensure that the pull request is assigned to the right base branch and that the development branch name contains the JIRA Task Id. Furthermore, each commit message should include the JIRA Task Id in the manner "ED-100: message".
- [ ]  Only update packages if it is mentioned and authorized in the design document, and make sure that you have the required permissions.
- [ ]  Avoid making API and database queries inside a loop as it can lead to performance issues and slow down the system.
- [ ]  When calling another function inside a given function, add comments explaining the purpose and meaning of the passed arguments and expected return values.
- [ ]  If adding a blank argument in a function, add a comment explaining the reason for the blank argument.
- [ ]  Before submitting a pull request, do a self-review of your code to ensure there are no conflicts with the base branch and all comments have been addressed.
- [ ]  Before merging a pull request, it's important to have other team members review it to catch any potential errors or issues
- [ ]  To maintain code integrity, it's important to remove all related changes when removing code during a code review.
- [ ]  If new constants, endpoints, or utility functions are introduced, it is important to check if they already exist in the service to avoid any duplication.
- [ ]  Whenever a new environment variable is added to a service, it's important to ensure that the necessary changes are made to related files such as ".env.sample" and "envVariables.js" to maintain consistency and avoid errors. Additionally, the new environment variable should be added to the devops repository to ensure that it is properly documented and accessible to the team.
- [ ]  When adding a new function to a service, it is important to document it with relevant information such as the name, parameters, and return value in a consistent format across all services. Additionally, if there are any changes to the API response, ensure that the documentation in the controllers is updated accordingly. 
- [ ]  Write a clear and concise commit message that describes the changes made.
- [ ]  Maintain consistent function signature and code across all services when adding a function to multiple services. Implement changes to database models in all services that use the same model. 
- [ ]  Use only let and const. Do not use var.
- [ ]  Make common functions for repetitive code blocks.
- [ ]  Avoid uploading sensitive information such as secret tokens or passwords in pull requests to ensure data security.
- [ ]  Maintain consistent indentation and spacing throughout the code.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,8 +24,8 @@ The Project building block facilitates the creation and engagement with micro-im
 
 # Supported Operating Systems
 
--   **Ubuntu (Recommended: Version 20 and above)** 
--   **Windows (Recommended: Version 11 and above)** 
+-   **Ubuntu (Recommended: Version 20 and above)**
+-   **Windows (Recommended: Version 11 and above)**
```
Comment (VISHNUDAS-tunerlabs) [inline README.md:28]: @MallanagoudaB all the readme changes you made is breaking the format. not accepting this. While commiting read me changes avoid eslint

File: documentation/1.0.0/native/envs/interface_env [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
-PROJECT_SERVICE_BASE_URL=http://localhost:5000
-ENTITY_SERVICE_BASE_URL=http://localhost:5001
-REQUIRED_BASE_PACKAGES=user,notification,scheduler,project,entity,survey
-ROUTE_CONFIG_JSON_URLS_PATHS="https://raw.githubusercontent.com/ELEVATE-Project/user/staging/src/constants/interface-routes/configs.json,https://raw.githubusercontent.com/ELEVATE-Project/project-service/main/constants/interface-routes/configs.json,https://raw.githubusercontent.com/ELEVATE-Project/samiksha-service/main/constants/interface-routes/configs.json,https://raw.githubusercontent.com/ELEVATE-Project/scheduler/develop/src/constants/interface-routes/configs.json,https://raw.githubusercontent.com/ELEVATE-Project/entity-management/main/src/constants/interface-routes/configs.json"
\ No newline at end of file
+API_DOC_URL= /interface/api-doc
+APPLICATION_ENV= development
+APPLICATION_PORT= 3569
+ENTITY_SERVICE_BASE_URL= http://localhost:5001
+INSTALLED_PACKAGES= elevate-user elevate-mentoring elevate-scheduler elevate-project elevate-entity-management elevate-self-creation-portal elevate-survey
```
Comment (VISHNUDAS-tunerlabs) [inline documentation/1.0.0/native/envs/interface_env:5]: @MallanagoudaB why we need mentoring , scp, and survey for project service?

File: documentation/1.0.0/native/envs/interface_env [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
+NOTIFICATION_SERVICE_BASE_URL= http://localhost:3002
+PROJECT_SERVICE_BASE_URL= http://localhost:5000
+RATE_LIMITER_ENABLED= false
+RATE_LIMITER_GENERAL_LIMIT= 50
+RATE_LIMITER_NUMBER_OF_PROXIES= 3
+RATE_LIMITER_PUBLIC_LOW_LIMIT= 5
+REQUIRED_BASE_PACKAGES= user,notification,scheduler,project,entity-management,self-creation-portal,survey
```
Comment (VISHNUDAS-tunerlabs) [inline documentation/1.0.0/native/envs/interface_env:12]: @MallanagoudaB why we need mentoring , scp, and survey for project service?

File: documentation/1.0.0/native/envs/interface_env [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
+REQUIRED_PACKAGES="elevate-user@1.1.77 elevate-scheduler@1.0.4 elevate-project@1.1.16 elevate-entity-management@1.0.10 elevate-survey@1.0.27"
```
Comment (VISHNUDAS-tunerlabs) [inline documentation/1.0.0/native/envs/interface_env:13]: why we need mentoring , scp, and survey for project service? @MallanagoudaB

File: documentation/1.0.0/native/envs/interface_env [hunk 5]
```diff
@@ -1,18 +1,19 @@
...
+ROUTE_CONFIG_JSON_URLS_PATHS="https://raw.githubusercontent.com/ELEVATE-Project/user/staging/src/constants/interface-routes/configs.json,https://raw.githubusercontent.com/ELEVATE-Project/project-service/main/constants/interface-routes/configs.json,https://raw.githubusercontent.com/ELEVATE-Project/samiksha-service/main/constants/interface-routes/configs.json,https://raw.githubusercontent.com/ELEVATE-Project/scheduler/develop/src/constants/interface-routes/configs.json,https://raw.githubusercontent.com/ELEVATE-Project/entity-management/main/src/constants/interface-routes/configs.json"
+SCHEDULER_SERVICE_BASE_URL= http://localhost:4000
+SELF_CREATION-PORTAL_SERVICE_BASE_URL= http://localhost:6001
```
Comment (VISHNUDAS-tunerlabs) [inline documentation/1.0.0/native/envs/interface_env:16]: @MallanagoudaB we don't need scp here

File: documentation/1.0.0/native/envs/interface_env [hunk 6]
```diff
@@ -1,18 +1,19 @@
...
+SUPPORTED_HTTP_TYPES= GET POST PUT PATCH DELETE
+SURVEY_SERVICE_BASE_URL= http://localhost:5007
```
Comment (VISHNUDAS-tunerlabs) [inline documentation/1.0.0/native/envs/interface_env:18]: We don't need survey service here @MallanagoudaB , check docker setup env of interface to get an idea

File: README.md [hunk 7]
```diff
@@ -716,12 +724,13 @@ Before setting up the following ELEVATE-Project application, dependencies given
     -   **Ubuntu/Linux**
 
         ```
-        pm2 start ./project-service/app.js --name project-service && \
-        pm2 start ./entity-management/src/app.js --name entity-management && \
-        pm2 start ./user/src/app.js --name user && \
-        pm2 start ./notification/src/app.js --name notification && \
-        pm2 start ./interface-service/src/app.js --name interface && \
-        pm2 start ./scheduler/src/app.js --name scheduler
+        (cd project-service && pm2 start app.js --name project-service && cd -) && \
```
Comment (VISHNUDAS-tunerlabs) [inline README.md:727]: @MallanagoudaB only this expected for native setup part in your documentation

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]
Reviewed on nov 3 2024

File: README.md (hunk 1 above)
Comment (MallanagoudaB) [reply README.md:28]: i didn't made any changes for this even i tried with new repo getting the same issue will check that correctly

File: README.md (hunk 7 above)
Comment (MallanagoudaB) [reply README.md:727]: yes @VISHNUDAS-tunerlabs  only this changes was made by me .

**Pull request closed without merging**